pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -560,3 +617,31 @@ func attrsCopy(attrs attributes) attributes {
 	}
 	return out
 }
+
+// GetName returns the plugin name
+func (mixerplugin) GetName() string {
+	return plugin.Mixer
+}
+
+// OnPrecompute implements the plugin interface
+func (mixerplugin) OnPrecompute(in *plugin.InputParams) interface{} {
+	out := &cacheData{
+		serviceConfigAny: make(map[cacheKey]*types.Any),
+	}
+	// TODO: unclear whether this enumerates all services, including external?
+	for _, service := range in.Push.ServiceByHostname {
+		for _, disable := range []bool{true, false} {
```

Comment:
instead of looping through `true` and `false` and then `inbound` and `outbound`, why not just write four calls ?

```
for _, service := range in.Push.ServiceByHostname {
   // policy disabled
   out.serviceConfigAny[cacheKey{"inbound", true, ... }] = util.MessageToAny(...)
   out.serviceConfigAny[cacheKey{"outbound", true, ... }] = util.MessageToAny(...)

   // policy enabled
   out.serviceConfigAny[cacheKey{"inbound", false, ... }] = util.MessageToAny(...)
   out.serviceConfigAny[cacheKey{"outbound", false, ... }] = util.MessageToAny(...)
}
```

seems more readable to me.

Reply:
Updated it. Does this look more reasonable? I'm trying to provision for more complicated keys (e.g. telemetry on/off options) and seems like a bunch of base cases is easier to maintain (similar to recommendation to use test case structs).